File: NHunspell/NHunspellExtender/GetSentencesEventHandler.cs
using System;

public delegate void GetSentencesEventHandler(object sender, GetSentenceEventArgs e);

public class GetSentenceEventArgs : EventArgs
{
    public string[] Sentences { get; set; }
    public string Text { get; }

    public GetSentenceEventArgs(string text)
    {
        this.Text = text;
    }
}

File: NHunspell/NHunspellExtender2/GetSentencesEventHandler.vb
Imports System.Windows.Forms

Public Delegate Sub GetSentencesEventHandler(sender As Object, e As GetSentenceEventArgs)

Public Class GetSentenceEventArgs
    Public Property Sentences As String()
    Public ReadOnly Property Text As String

    Public Sub New(text As String)
        Me.Text = text
    End Sub

End Class
